diff --git a/scripts/Engineering/EnggUtils.py b/scripts/Engineering/EnggUtils.py
index fc9f90cb764..df0b14a5fde 100644
--- a/scripts/Engineering/EnggUtils.py
+++ b/scripts/Engineering/EnggUtils.py
@@ -10,7 +10,7 @@ from mantid.kernel import IntArrayProperty, UnitConversion, DeltaEModeType, logg
 import mantid.simpleapi as mantid
 from mantid.simpleapi import AnalysisDataService as ADS
 from matplotlib import gridspec
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
+from Engineering.common import path_handling
 
 ENGINX_BANKS = ['', 'North', 'South', 'Both: North, South', '1', '2']
 
diff --git a/scripts/Engineering/gui/engineering_diffraction/presenter.py b/scripts/Engineering/gui/engineering_diffraction/presenter.py
index c9b7d3c8396..cc4aff027ea 100644
--- a/scripts/Engineering/gui/engineering_diffraction/presenter.py
+++ b/scripts/Engineering/gui/engineering_diffraction/presenter.py
@@ -5,7 +5,7 @@
 #   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 # SPDX - License - Identifier: GPL - 3.0 +
 
-from .tabs.common import path_handling
+from Engineering.common import path_handling
 from .tabs.common import CalibrationObserver
 from .tabs.calibration.model import CalibrationModel
 from .tabs.calibration.view import CalibrationView
diff --git a/scripts/Engineering/gui/engineering_diffraction/settings/settings_model.py b/scripts/Engineering/gui/engineering_diffraction/settings/settings_model.py
index fb003562f72..5ecc877a908 100644
--- a/scripts/Engineering/gui/engineering_diffraction/settings/settings_model.py
+++ b/scripts/Engineering/gui/engineering_diffraction/settings/settings_model.py
@@ -5,7 +5,7 @@
 #   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 # SPDX - License - Identifier: GPL - 3.0 +
 from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting, set_setting
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
+from Engineering.common import path_handling
 
 
 class SettingsModel(object):
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py b/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py
index 16bd9160570..9cc2f5848d0 100644
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py
+++ b/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py
@@ -13,7 +13,7 @@ from mantid.simpleapi import PDCalibration, DeleteWorkspace, CloneWorkspace, Dif
 import Engineering.EnggUtils as EnggUtils
 from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting, set_setting
 from Engineering.gui.engineering_diffraction.tabs.common import vanadium_corrections
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
+from Engineering.common import path_handling
 
 CALIB_PARAMS_WORKSPACE_NAME = "engggui_calibration_banks_parameters"
 
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/presenter.py b/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/presenter.py
index 388ecfff0cc..b0e2f085183 100644
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/presenter.py
+++ b/scripts/Engineering/gui/engineering_diffraction/tabs/calibration/presenter.py
@@ -7,11 +7,11 @@
 # pylint: disable=invalid-name
 from copy import deepcopy
 
+from Engineering.common import path_handling
 from Engineering.gui.engineering_diffraction.tabs.common import INSTRUMENT_DICT, create_error_message
 from Engineering.gui.engineering_diffraction.tabs.common.calibration_info import CalibrationInfo
 from Engineering.gui.engineering_diffraction.tabs.common.cropping.cropping_presenter import CroppingPresenter
 from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting, set_setting
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
 
 from mantidqt.utils.asynchronous import AsyncTask
 from mantid.simpleapi import logger
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/common/path_handling.py b/scripts/Engineering/gui/engineering_diffraction/tabs/common/path_handling.py
deleted file mode 100644
index 97fdbd900fb..00000000000
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/common/path_handling.py
+++ /dev/null
@@ -1,33 +0,0 @@
-# Mantid Repository : https://github.com/mantidproject/mantid
-#
-# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
-#   NScD Oak Ridge National Laboratory, European Spallation Source,
-#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
-# SPDX - License - Identifier: GPL - 3.0 +
-from os import path
-from mantid.kernel import logger
-from mantid.simpleapi import Load
-from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
-
-INTERFACES_SETTINGS_GROUP = "CustomInterfaces"
-ENGINEERING_PREFIX = "EngineeringDiffraction2/"
-
-
-def get_run_number_from_path(run_path, instrument):
-    return path.splitext(path.basename(run_path))[0].replace(instrument, '').lstrip('0')
-
-
-def get_output_path():
-    location = get_setting(INTERFACES_SETTINGS_GROUP, ENGINEERING_PREFIX, "save_location")
-    return location if location is not None else ""
-
-
-def load_workspace(file_path):
-    try:
-        return Load(Filename=file_path, OutputWorkspace="engggui_calibration_sample_ws")
-    except Exception as e:
-        logger.error("Error while loading workspace. "
-                     "Could not run the algorithm Load successfully for the data file "
-                     "(path: " + str(file_path) + "). Error description: " + str(e)
-                     + " Please check also the previous log messages for details.")
-        raise RuntimeError
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py b/scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py
index 96e256c3d11..79ca51f1652 100644
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py
+++ b/scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py
@@ -10,7 +10,7 @@ from mantid.simpleapi import logger, Load, SaveNexus, NormaliseByCurrent, Integr
     ApplyDiffCal, ConvertUnits
 from mantid.simpleapi import AnalysisDataService as Ads
 
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
+from Engineering.common import path_handling
 from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
 
 VANADIUM_INPUT_WORKSPACE_NAME = "engggui_vanadium_ws"
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py b/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py
index 28abe70e3c6..b3779975285 100644
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py
+++ b/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py
@@ -9,7 +9,7 @@ from os import path
 from mantid.simpleapi import Load, logger, EnggEstimateFocussedBackground, ConvertUnits, Minus, AverageLogData, \
     CreateEmptyTableWorkspace, GroupWorkspaces, DeleteWorkspace, DeleteTableRows, RenameWorkspace, CreateWorkspace
 from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
+from Engineering.common import path_handling
 from mantid.api import AnalysisDataService as ADS
 from mantid.api import TextAxis
 from mantid.kernel import UnitConversion, DeltaEModeType, UnitParams
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/EngDiff_fitpropertybrowser.py b/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/EngDiff_fitpropertybrowser.py
index 5b1fcf4afc0..2f87da10216 100644
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/EngDiff_fitpropertybrowser.py
+++ b/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/EngDiff_fitpropertybrowser.py
@@ -13,8 +13,8 @@ from mantidqt.utils.qt import import_qt
 from mantidqt.utils.observer_pattern import GenericObservable
 from mantidqt.widgets.fitpropertybrowser import FitPropertyBrowser
 from mantid.api import AnalysisDataService as ADS
+from Engineering.common import path_handling
 from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
 
 BaseBrowser = import_qt('.._common', 'mantidqt.widgets', 'FitPropertyBrowser')
 
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py b/scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py
index 8027a336e17..568e3c8973c 100644
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py
+++ b/scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py
@@ -9,7 +9,8 @@ from os import path, makedirs
 from matplotlib import gridspec
 import matplotlib.pyplot as plt
 
-from Engineering.gui.engineering_diffraction.tabs.common import vanadium_corrections, path_handling
+from Engineering.common import path_handling
+from Engineering.gui.engineering_diffraction.tabs.common import vanadium_corrections
 from Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
 from Engineering import EnggUtils
 from mantid.simpleapi import logger, AnalysisDataService as Ads, SaveNexus, SaveGSS, SaveFocusedXYE, \
diff --git a/scripts/Engineering/gui/engineering_diffraction/tabs/focus/test/test_focus_model.py b/scripts/Engineering/gui/engineering_diffraction/tabs/focus/test/test_focus_model.py
index e9ba3233009..4a8330abd8c 100644
--- a/scripts/Engineering/gui/engineering_diffraction/tabs/focus/test/test_focus_model.py
+++ b/scripts/Engineering/gui/engineering_diffraction/tabs/focus/test/test_focus_model.py
@@ -11,8 +11,8 @@ from os import path
 
 from unittest.mock import patch, MagicMock, call
 from mantid.simpleapi import CreateSampleWorkspace
+from Engineering.common import path_handling
 from Engineering.gui.engineering_diffraction.tabs.focus import model
-from Engineering.gui.engineering_diffraction.tabs.common import path_handling
 from Engineering.gui.engineering_diffraction.tabs.common.calibration_info import CalibrationInfo
 
 file_path = "Engineering.gui.engineering_diffraction.tabs.focus.model"
