diff --git a/Framework/Kernel/CMakeLists.txt b/Framework/Kernel/CMakeLists.txt
index 2bea0a52e63..56f6e825f2e 100644
--- a/Framework/Kernel/CMakeLists.txt
+++ b/Framework/Kernel/CMakeLists.txt
@@ -489,6 +489,12 @@ set_target_properties(Kernel
                                  COMPILE_DEFINITIONS
                                  "IN_MANTID_KERNEL;PSAPI_VERSION=1")
 
+target_include_directories(Kernel PUBLIC 		
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
+		$<INSTALL_INTERFACE:include/Mantid>
+)
+
+
 target_include_directories(Kernel SYSTEM
                            PUBLIC ${Boost_INCLUDE_DIRS} ${POCO_INCLUDE_DIRS}
                                   ${JSONCPP_INCLUDE_DIR} ${NEXUS_INCLUDE_DIR}
@@ -607,7 +613,7 @@ endif()
 
 # Installation settings
 
-mtd_install_targets(TARGETS Kernel INSTALL_DIRS ${LIB_DIR} ${WORKBENCH_LIB_DIR})
+mtd_install_dylib(TARGETS Kernel INSTALL_DIRS ${LIB_DIR} ${WORKBENCH_LIB_DIR})
 
 # Create the properties file for the installer
 set(MANTID_ROOT_BUILD ${MANTID_ROOT})
diff --git a/buildconfig/CMake/TargetFunctions.cmake b/buildconfig/CMake/TargetFunctions.cmake
index 743a090ebbe..e5b82760899 100644
--- a/buildconfig/CMake/TargetFunctions.cmake
+++ b/buildconfig/CMake/TargetFunctions.cmake
@@ -15,13 +15,43 @@ function (mtd_install_targets)
         message(FATAL_ERROR "Empty argument TARGETS")
         return()
     endif()
+    _sanitize_install_dirs(_install_dirs ${PARSED_INSTALL_DIRS})
+    foreach( _dir ${_install_dirs})
+        install ( TARGETS ${PARSED_TARGETS} ${SYSTEM_PACKAGE_TARGET} DESTINATION ${_dir} )
+    endforeach ()
+endfunction()
+
 
+# Install a target into multiple directories
+# This respects ENABLE_WORKBENCH flags for macOS
+function (mtd_install_dylib)
+  set (options)
+  set (oneValueArgs TARGETS)
+  set (multiValueArgs INSTALL_DIRS)
+  cmake_parse_arguments (PARSED "${options}" "${oneValueArgs}"
+                         "${multiValueArgs}" ${ARGN})
+
+    if (NOT PARSED_INSTALL_DIRS)
+        message(FATAL_ERROR "Empty argument INSTALL_DIRS")
+        return()
+    endif()
+    if (NOT PARSED_TARGETS)
+        message(FATAL_ERROR "Empty argument TARGETS")
+        return()
+    endif()
+    install(
+        DIRECTORY inc/
+        DESTINATION include/Mantid
+        COMPONENT Devel
+        PATTERN ".svn" EXCLUDE
+    )
     _sanitize_install_dirs(_install_dirs ${PARSED_INSTALL_DIRS})
     foreach( _dir ${_install_dirs})
         install ( TARGETS ${PARSED_TARGETS} ${SYSTEM_PACKAGE_TARGET} DESTINATION ${_dir} )
     endforeach ()
 endfunction()
 
+
 # Install files into multiple directories
 # This respects ENABLE_WORKBENCH flags for macOS
 function (mtd_install_files)
